Public/Src/Cache/ContentStore/Distributed/Utilities/GrpcDistributedPathTransformer.cs

```diff
@@ -50,31 +51,30 @@ public byte[] GetLocalMachineLocation(AbsolutePath cacheRoot)
                 cacheRoot = cacheRoot / Constants.SharedDirectoryName;
             }
 
-            var cacheRootString = cacheRoot.Path.ToUpperInvariant();
+            var cacheRootString = cacheRoot.Path.PathToUpperInvariant();
 
             // Determine if cacheRoot needs to be accessed through its directory junction
             var directories = _junctionsByDirectory.Keys;
             var directoryToReplace = directories.SingleOrDefault(directory =>
-                                        cacheRootString.StartsWith(directory, StringComparison.OrdinalIgnoreCase));
+                                        cacheRootString.StartsWith(directory, OperatingSystemHelper.IsWindowsOS ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal));

```

Comment:
In all our current macOS builds the file system is configured to be case insensitive.

Reply:
That's great! I've updated the PR based on this.

---
In reply to: [281299637](https://github.com/Microsoft/BuildXL/pull/253#discussion_r281299637) [](ancestors = 281299637)